Detectors/AOD/src/StandaloneAODProducer.cxx

```diff
@@ -0,0 +1,169 @@
+        //DECLARE_SOA_TABLE(Collisions, "AOD", "COLLISION", o2::soa::Index<>,
+        // collision::BCId, collision::PosX, collision::PosY, collision::PosZ,
+        // collision::CovXX, collision::CovXY, collision::CovXZ, collision::CovYY, collision::CovYZ, collision::CovZZ,
+        // collision::Chi2, collision::NumContrib, collision::CollisionTime, collision::CollisionTimeRes, collision::CollisionTimeMask);
+        int BCid = 0;
+        auto& cov = v.getCov();
+        auto& ts = v.getTimeStamp();
+
+        // TODO: figure out BC + CollisionTimeMask
+        collCursor(0, BCid, v.getX(), v.getY(), v.getZ(),
+                   cov[0], cov[1], cov[2], cov[3], cov[4], cov[6],
+                   v.getChi2(), v.getNContributors(), ts.getTimeStamp(), ts.getTimeStampError(), 1);
+
+        // get the track for each vertex and fill the tracks table
+        // now go over tracks via the indices
+        auto& trackref = (*v2trackref)[index];
+        int start = trackref.getFirstEntryOfSource(0);
+        int ntracks = trackref.getEntriesOfSource(0);
+        for (int ti = 0; ti < ntracks; ++ti) {
+          auto trackindex = (*vertexTrackIDs)[start + ti];
+
+          // now we need to fetch the actual track and fill the table
+          const auto source = trackindex.getSource();
+          o2::track::TrackParCov* track = nullptr;
+          if (source == o2::dataformats::VtxTrackIndex::Source::TPC) {
+            track = &((*tpctracks)[trackindex.getIndex()]);
+          } else if (source == o2::dataformats::VtxTrackIndex::Source::ITS) {
+            track = &((*itstracks)[trackindex.getIndex()]);
+          } else if (source == o2::dataformats::VtxTrackIndex::Source::TPCITS) {
+            track = &((*itstpctracks)[trackindex.getIndex()]);
+          } else {
+            LOG(WARNING) << "Unsupported track source";
+          }
+
+          //DECLARE_SOA_TABLE_FULL(StoredTracks, "Tracks", "AOD", "TRACK:PAR",
+          //                       o2::soa::Index<>, track::CollisionId, track::TrackType,
+          //                       track::X, track::Alpha,
+          //                       track::Y, track::Z, track::Snp, track::Tgl,
+          //                       track::Signed1Pt,
+          //                       track::NormalizedPhi<track::RawPhi>,
+          //                       track::Px<track::Signed1Pt, track::Snp, track::Alpha>,
+          //                       track::Py<track::Signed1Pt, track::Snp, track::Alpha>,
+          //                       track::Pz<track::Signed1Pt, track::Tgl>,
+          //                      track::Charge<track::Signed1Pt>);
+
+          std::array<float, 3> pxpypz;
+          track->getPxPyPzGlo(pxpypz);
+          trackCursor(0, index, 0 /* CORRECT THIS */, track->getX(), track->getAlpha(), track->getY(), track->getZ(), track->getSnp(), track->getTgl(),
+                      track->getPt() /*CHECK!!*/, track->getPhi(), pxpypz[0], pxpypz[1], pxpypz[2]);
```

Comment:
I understand that this constructor is intended for aliroot AOD tracks conversion, for O2 we should not go via lab momentum.